src/crud.py

```
@@ -0,0 +1,68 @@
+from fastapi import HTTPException
+from sqlalchemy.ext.asyncio import AsyncSession
+from sqlalchemy import select
+from sqlalchemy.orm import joinedload
+
+from database.models import MovieModel
+
+
+async def get_movies(db: AsyncSession, per_page: int, page: int):
+    offset = (page - 1) * per_page
+    result = await db.execute(
+        select(MovieModel)
+        .offset(offset)
+        .limit(per_page)
+        .order_by(MovieModel.id.desc())
+    )
+    movies = result.scalars().all()
+    if not movies:
+        raise HTTPException(status_code=404, detail="No movies found.")
```

Design note: `get_movies` raises `HTTPException(status_code=404, detail="No movies found.")` inside the CRUD layer. This couples DB layer to HTTP semantics. It's functional, but for clearer separation of concerns you might return an empty list (or None) and let the route raise the exact `HTTPException` with the required message. If you keep this approach, ensure the message exactly matches the spec (it does here).